interfaces/hotplug/deviceinfo.go

```diff
@@ -89,3 +89,38 @@ func (h *HotplugDeviceInfo) Attribute(name string) (string, bool) {
 	val, ok := h.Data[name]
 	return val, ok
 }
+
+func (h *HotplugDeviceInfo) firstAttrValueOf(tryAttrs ...string) string {
+	for _, attr := range tryAttrs {
+		if val, _ := h.Attribute(attr); val != "" {
+			return val
+		}
+	}
+	return ""
+}
+
+func (h *HotplugDeviceInfo) String() string {
+	var nameOrPath string
+
+	// devname is the name of the device under /dev, eg. /dev/ttyS0;
+	// prefer devname over devpath as this is the one used to talk to the device.
+	if nameOrPath = h.DeviceName(); nameOrPath == "" {
+		// devpath is the path of the device under /sys, eg. /sys/devices/pnp0/00:04/tty/ttyS0.
+		nameOrPath = h.DevicePath()
+	}
+
+	modelOrVendor := h.firstAttrValueOf("ID_MODEL_FROM_DATABASE", "ID_MODEL", "ID_MODEL_ID", "ID_VENDOR_FROM_DATABASE", "ID_VENDOR", "ID_VENDOR_ID")
+	if modelOrVendor == "" {
+		modelOrVendor = "?"
+	}
+	if len(modelOrVendor) > 16 {
+		modelOrVendor = modelOrVendor[0:16] + "..."
+	}
+
+	serial := h.firstAttrValueOf("ID_SERIAL_SHORT", "ID_SERIAL")
+	if serial == "noserial" || serial == "" {
+		serial = "?"
```

Comment:
same wondering

Reply:
should we reverse our preference to ID_SERIAL if there was no model or vendor?

Reply:
Good idea, done.